### PR TITLE
Implement most methods related to getHighestBlock(Y)At

### PR DESCRIPTION
**Pull request by Thorinwasher**
# Description
Implement the following methods in WorldMock:
- getHighestBlockYAt(int,int)
- getHighestBlockYat(Location)
- getHighestBlockAt(int,int)
- getHighestBlockAt(Location)
- getHighestBlockYAt(int,int,HeightMap)
- getHighestBlockYAt(Location,HeightMap)
- getHighestBlockAt(int,int,HeightMap)
- getHighestBlockAt(Location,HeightMap

BlockMock#isSolid() is not implemented yet, that makes 4 of these tests to skip.

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Code follows existing style.
- [x] Unit tests added (if applicable).

### CODE REVIEW

**Comment by thelooter** [comment]
I don't think it makes sense to merge this until `BlockMock#isSolid()` is implemented

File: src/main/java/be/seeseemelk/mockbukkit/WorldMock.java [hunk 1]
```diff
@@ -75,33 +75,13 @@
...
-import org.bukkit.Particle;
-import org.bukkit.Raid;
-import org.bukkit.Sound;
-import org.bukkit.SoundCategory;
-import org.bukkit.StructureType;
-import org.bukkit.TreeType;
-import org.bukkit.World;
-import org.bukkit.WorldCreator;
-import org.bukkit.WorldType;
+import org.bukkit.*;
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/WorldMock.java]: No wildcard imports.

File: src/main/java/be/seeseemelk/mockbukkit/WorldMock.java [hunk 2]
```diff
@@ -2582,32 +2552,71 @@ public boolean createExplosion(Location loc, float power, boolean setFire, boole
...
+		return this.getMinHeight() - 1;
+	}
+
+	private boolean isOfHeightMap(Block block, HeightMap heightMap)
+	{
+		return switch (heightMap)
+		{
+			case MOTION_BLOCKING ->
+			{
+				boolean isWaterLogged = false;
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/WorldMock.java]:
```suggestion
boolean isWaterLogged = block.getBlockData() instanceof Waterlogged waterlogged && waterlogged.isWaterlogged();

File: src/main/java/be/seeseemelk/mockbukkit/WorldMock.java [hunk 3]
```diff
@@ -2582,32 +2552,71 @@ public boolean createExplosion(Location loc, float power, boolean setFire, boole
...
+				if (block.getBlockData() instanceof Waterlogged waterlogged)
+				{
+					isWaterLogged = waterlogged.isWaterlogged();
+				}
+				yield (block.isSolid() || isWaterLogged || block.isLiquid()) && !Tag.LEAVES.isTagged(block.getType());
+			}
+			case OCEAN_FLOOR -> block.isSolid();
+			case OCEAN_FLOOR_WG, WORLD_SURFACE_WG -> throw new UnimplementedOperationException();
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/WorldMock.java]: What's the condition for these? Can we implement them?

**Review by Insprill — DISMISSED** [review]

File: src/main/java/be/seeseemelk/mockbukkit/WorldMock.java (hunk 1 above)
Comment (Thorinwasher) [reply src/main/java/be/seeseemelk/mockbukkit/WorldMock.java]: Alright, that's my auto formatting. Let me see if I can fix that. I dislike them as well

File: src/main/java/be/seeseemelk/mockbukkit/WorldMock.java (hunk 3 above)
Comment (Thorinwasher) [reply src/main/java/be/seeseemelk/mockbukkit/WorldMock.java]: WG, I think that stands for world generation. So this has to relate to how the world was generated. Should be doable as the worlds are quite simple (just flat worlds)

File: src/main/java/be/seeseemelk/mockbukkit/WorldMock.java (hunk 1 above)
Comment (thelooter) [reply src/main/java/be/seeseemelk/mockbukkit/WorldMock.java]: Go to IntelliJ Settings > Editor > Code Style > Java > Imports and set both
- Class count to use import with .*
- Names count to use import with .*

to an insanely high number like 99999999

**Review by thelooter — APPROVED** [review]
LGTM

**Pull request merged**